lib/network/http2_network_test.go

```diff
@@ -0,0 +1,174 @@
+package sebaknetwork
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"unicode"
+
+	"boscoin.io/sebak/lib/common"
+	"github.com/stellar/go/keypair"
+	"github.com/stretchr/testify/assert"
+)
+
+func ExampleHttp2NetworkConfigCreateWithNonTLS() {
+	var config HTTP2NetworkConfig
+	endpoint, err := sebakcommon.NewEndpointFromString("https://localhost:5000?NodeName=n1")
+	if err != nil {
+		fmt.Print("Error in NewEndpointFromString")
+	}
+	queries := endpoint.Query()
+	queries.Add("TLSCertFile", "")
+	queries.Add("TLSKeyFile", "")
+	endpoint.RawQuery = queries.Encode()
+
+	config, err = NewHTTP2NetworkConfigFromEndpoint(endpoint)
+	if err != nil {
+		fmt.Print("Error in NewHTTP2NetworkConfigFromEndpoint")
+	}
+	fmt.Println(config.NodeName)
+	fmt.Println(config.Addr)
+
+	// Output: n1
+	// localhost:5000
+}
+
+const (
+	dirPath  = "tmp"
+	certPath = "cert.pem"
+	keyPath  = "key.pem"
+)
+
+func createNewHTTP2Network(t *testing.T) (kp *keypair.Full, mn *HTTP2Network, validator *sebakcommon.Validator) {
+	g := NewKeyGenerator(dirPath, certPath, keyPath)
+
+	var config HTTP2NetworkConfig
+	endpoint, err := sebakcommon.NewEndpointFromString("https://localhost:5000?NodeName=n1")
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	queries := endpoint.Query()
+	queries.Add("TLSCertFile", g.GetCertPath())
+	queries.Add("TLSKeyFile", g.GetKeyPath())
+	endpoint.RawQuery = queries.Encode()
+
+	config, err = NewHTTP2NetworkConfigFromEndpoint(endpoint)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	mn = NewHTTP2Network(config)
+
+	kp, _ = keypair.Random()
+	validator, _ = sebakcommon.NewValidator(kp.Address(), mn.Endpoint(), "")
+	validator.SetKeypair(kp)
+
+	mn.SetContext(context.WithValue(context.Background(), "currentNode", validator))
+
+	return
+}
+
+type TestMessageBroker struct{}
+
+func (r TestMessageBroker) ResponseMessage(w http.ResponseWriter, o string) {
+	fmt.Fprintf(w, o)
+}
```

Comment:
Is it same with `TestMessageBroker.ResponseMessage`. Why is it need? :->
```
+func (r Http2MessageBroker) ResponseMessage(w http.ResponseWriter, o string) {
+	fmt.Fprintf(w, string(o))
+}
```

Reply:
I added a new testcase(TestHTTP2NetworkMessageBrokerResponseMessage) using ResponseMessage(). Doesn't it make sense?